pkg/querier/querier.go

```diff
@@ -46,6 +47,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.Iterators, "querier.iterators", false, "Use iterators to execute query, as opposed to fully materialising the series in memory.")
 	f.BoolVar(&cfg.BatchIterators, "querier.batch-iterators", false, "Use batch iterators to execute query, as opposed to fully materialising the series in memory.  Takes precedent over the -querier.iterators flag.")
 	f.BoolVar(&cfg.IngesterStreaming, "querier.ingester-streaming", false, "Use streaming RPCs to query ingester.")
+	f.BoolVar(&cfg.TSDBIngesterStreaming, "querier.tsdb-ingester-streaming", false, "Use TSDB streaming RPCs to query ingester.")
```

Comment:
I'm missing the rationale on adding a new flag instead of reusing the `cfg.IngesterStreaming`. Is there any specific reason?

Reply:
Since there's a separate path for TSDB streaming I needed a way to  select that path, and we didn't pass in the storage config to the querier. But to reuse it, that's what we'd have to do. I don't have a strong preference.

Reply:
I would have preferred if streaming API call handled both cases, perhaps extending `QueryStreamResponse` with `TimeSeries` field, and then selecting appropriate path (chunks-based, or timeseries-based) in the querier. To me TSDB or Blocks setting should only affect ingesters. (See also discussion at proto file)

Reply:
This config option is removed.